File: mpi/mpi-c/collect-sync.c
#include <stdio.h>
#include "mpi.h"

int main(int argv, char* argc[]){

    MPI_Init(&argv, &argc);

    int rank, numprocs;

    MPI_Comm_size(MPI_COMM_WORLD, &numprocs);
    MPI_Comm_rank(MPI_COMM_WORLD, &rank);

    for(int i = 0; i < numprocs; i++){
        if(rank == i){
            printf("Hello World from process %d of %d\n", rank, numprocs);
        }
        MPI_Barrier(MPI_COMM_WORLD);
    }

    MPI_Finalize();
}

File: mpi/mpi-fort/collect-sync.f90
PROGRAM hello_world_mpi
include 'mpif.h'

integer rank, numprocs, ierr

CALL MPI_INIT(ierr)
CALL MPI_COMM_SIZE(MPI_COMM_WORLD, numprocs, ierr)
CALL MPI_COMM_RANK(MPI_COMM_WORLD, rank, ierr)

DO i = 0, numprocs, 1
    IF(rank == i) THEN
        print *, 'Hello World from process: ', rank, 'of ', numprocs
    END IF
    CALL MPI_BARRIER(MPI_COMM_WORLD, ierr)
END DO

CALL MPI_FINALIZE(ierr)
END PROGRAM
